### PR TITLE
CoreLib Build System portability updates

### PR DESCRIPTION
**Pull request by Rew**
Updates to implement #10836

### CODE REVIEW

File: src/mscorlib/shared/System/Diagnostics/Tracing/TraceLogging/TraceLoggingEventSource.cs [hunk 1]
```diff
@@ -5,13 +5,13 @@
 // This program uses code hyperlinks available as part of the HyperAddin Visual Studio plug-in.
 // It is available from http://www.codeplex.com/hyperAddin 
 
-#if !PLATFORM_UNIX
+#if PLATFORM_WINDOWS
```
Comment (jkotas) [inline src/mscorlib/shared/System/Diagnostics/Tracing/TraceLogging/TraceLoggingEventSource.cs]: I am not sure whether the Windows builds defines PLATFORM_WINDOWS. May need to add it.

**Comment by jkotas** [comment]
Could you please resolve the merge conflicts?

**Comment by Rew** [comment]
Sorry about the confilicts, I wasn't paying attention last night and forgot to sync with the master.  It is merged now.

You were correct that PLATFORM_WINDOWS was not defined.  While adding the define for it I realized I had missed a few more places that it should be used.  All updated now.

File: src/mscorlib/System.Private.CoreLib.csproj [hunk 2]
```diff
@@ -734,10 +692,44 @@
   <ItemGroup>
     <Compile Include="$(BclSourcesRoot)\System\Numerics\Hashing\HashHelpers.cs" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetsUnix)' != 'true'">
+  <ItemGroup Condition="'$(TargetsUnix)' == 'true'">
+    <Compile Include="$(BclSourcesRoot)\Interop\Unix\System.Globalization.Native\Interop.Collation.cs" />
+    <Compile Include="$(BclSourcesRoot)\Interop\Unix\System.Globalization.Native\Interop.ICU.cs" />
+	  <Compile Include="$(BclSourcesRoot)\System\Diagnostics\Debug.Unix.cs" />
```
Comment (jkotas) [inline src/mscorlib/System.Private.CoreLib.csproj]: Nit: indentation

File: src/mscorlib/System.Private.CoreLib.csproj [hunk 3]
```diff
@@ -734,10 +692,44 @@
...
+    <Compile Include="$(BclSourcesRoot)\System\Globalization\EncodingDataItem.Unix.cs" />
+    <Compile Include="$(BclSourcesRoot)\System\Globalization\GlobalizationMode.Unix.cs" />
+    <Compile Include="$(BclSourcesRoot)\System\Globalization\HijriCalendar.Unix.cs" />
+    <Compile Include="$(BclSourcesRoot)\System\Globalization\IdnMapping.Unix.cs" />
+    <Compile Include="$(BclSourcesRoot)\System\Globalization\JapaneseCalendar.Unix.cs" />
+    <Compile Include="$(BclSourcesRoot)\System\Globalization\TextInfo.Unix.cs" />
+    <Compile Include="$(BclSourcesRoot)\System\IO\FileSystemEnumerable.cs" />
+    <Compile Include="$(BclSourcesRoot)\System\IO\TextReader.cs" />
+    <Compile Include="$(BclSourcesRoot)\System\IO\StreamReader.cs" />
+	  <Compile Include="$(BclSourcesRoot)\System\Runtime\Versioning\CompatibilitySwitch.cs" />
```
Comment (jkotas) [inline src/mscorlib/System.Private.CoreLib.csproj]: Nit: indentiation

**Comment by jkotas** [comment]
Thanks!

**Pull request merged**